src/Conversion/ONNXToKrnl/ControlFlow/Loop.cpp

```diff
@@ -31,7 +31,11 @@ struct ONNXLoopOpLowering : public ConversionPattern {
       ConversionPatternRewriter &rewriter) const final {
     auto loc = ONNXLoc<ONNXLoopOp>(op);
     auto loopOp = dyn_cast<ONNXLoopOp>(op);
-    ONNXLoopOpAdaptor loopOpAdapter(operands, op->getAttrDictionary());
+    ONNXLoopOpAdaptor loopOpAdaptor(operands, op->getAttrDictionary());
+
+    if (needWhile(op)) {
+      return rewriteWithWhile(op, operands, rewriter);
```

Comment:
Would it be better to use SCFWhile instead of While, e.g `needSCFWhile`, `rewriteWithSCFWhile`?